discovery-infra/test_infra/helper_classes/cluster.py

```diff
@@ -545,3 +581,39 @@ def reset_cluster_and_wait_for_ready(self, cluster, nodes):
         cluster.wait_until_hosts_are_discovered()
         cluster.wait_for_ready_to_install()
 
+    def get_events(self, host_id=''):
+        return self.api_client.get_events(cluster_id=self.id, host_id=host_id)
+
+    def find_event(self, event_to_find, reference_time, params_list=[], host_id=''):
+        events_list = self.get_events(host_id=host_id)
+        event_exist = False
+        for event in events_list:
+            if event_to_find in event['message']:
+                if utils.to_utc(event['event_time']) > reference_time:
+                    # For debugging
+                    logging.info(f"Event time: {utils.to_utc(event['event_time'])} ")
+                    logging.info(f"Reference time: {reference_time} ")
+                    if all(param in event['message'] for param in params_list):
+                        event_exist = True
+                        logging.info(f"Event to find: {event_to_find} exists with its params")
+                        return event_exist
+        return event_exist
+
+    def wait_for_event(self, event_to_find, reference_time, params_list=[], host_id=''):
+        logging.info(f"Searching for event: {event_to_find}")
+        try:
+            waiting.wait(
+                lambda: self.find_event(
+                    event_to_find,
+                    reference_time,
+                    params_list,
+                    host_id
+                ),
+                timeout_seconds=30,
+                sleep_seconds=2,
+                waiting_for="Event: %s" % event_to_find,
+            )
+        except:
```

Comment:
except on the specific exception `waiting.exceptions.TimeoutExpired`

Reply:
Done, please review